crates/cast/src/starknet_commands/script.rs

```diff
@@ -0,0 +1,341 @@
+
+    let runner = SierraCasmRunner::new(
+        sierra_program,
+        Some(MetadataComputationConfig::default()),
+        OrderedHashMap::default(),
+    )
+    .with_context(|| "Failed setting up runner.")?;
+
+    let name_suffix = module_name.to_string() + "::main";
+    let func = runner.find_function(name_suffix.as_str())?;
+
+    let (entry_code, builtins) = runner.create_entry_code(func, &Vec::new(), usize::MAX)?;
+    let footer = runner.create_code_footer();
+    let instructions = chain!(
+        entry_code.iter(),
+        runner.get_casm_program().instructions.iter(),
+        footer.iter()
+    );
+
+    // import from cairo-lang-runner
+    let (hints_dict, string_to_hint) = build_hints_dict(instructions.clone());
+
+    let mut cairo_hint_processor = CairoHintProcessor {
+        hints: &string_to_hint,
+        provider,
+        runtime,
+        run_resources: RunResources::default(),
+    };
+
+    match runner.run_function(
+        func,
+        &mut cairo_hint_processor,
+        hints_dict,
+        instructions,
+        builtins,
+    ) {
+        Ok(result) => match result.value {
+            RunResultValue::Success(data) => Ok(ScriptResponse {
+                status: "success".to_string(),
+                msg: build_readable_text(&data),
+            }),
+            RunResultValue::Panic(panic_data) => Ok(ScriptResponse {
+                status: "script panicked".to_string(),
+                msg: build_readable_text(&panic_data),
+            }),
+        },
+        Err(err) => Err(err.into()),
+    }
+}
+
+fn compile_script(path_to_scarb_toml: Option<Utf8PathBuf>) -> Result<Utf8PathBuf> {
+    let scripts_manifest_path = match path_to_scarb_toml {
+        Some(path) => path,
+        None => get_scarb_manifest()
+            .context("Failed to obtain manifest path from scarb")
+            .unwrap(),
+    };
+    ensure!(
+        scripts_manifest_path.exists(),
+        "Path {scripts_manifest_path} does not exist"
+    );
+
+    ScarbCommand::new()
+        .arg("build")
+        .env("SCARB_MANIFEST_PATH", &scripts_manifest_path)
+        .run()?;
+
+    let opts = ScarbOpts { with_deps: true };
+    let metadata = get_scarb_metadata(&scripts_manifest_path, opts)?;
+    let package_metadata = get_package_metadata(&metadata, &scripts_manifest_path)?;
+
+    let filename = format!("{}.sierra.json", package_metadata.name);
+    let path = metadata
+        .target_dir
+        .unwrap_or(metadata.workspace.root.join("target"))
+        .join(metadata.current_profile)
+        .join(filename.clone());
+
+    ensure!(
+        path.exists(),
+        "package has not been compiled, file does not exist: {path}"
+    );
+
+    Ok(path)
+}
+
+// taken from starknet-foundry/crates/forge/src/test_case_summary.rs
+/// Helper function to build `readable_text` from a run data.
+fn build_readable_text(data: &Vec<Felt252>) -> Option<String> {
```

Comment:
Let's create an issue to move this code to a common crate

Reply:
https://github.com/foundry-rs/starknet-foundry/issues/1127